### PR TITLE
#1077 Change the streamApp java api generic type <K, V> to Row

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1077)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-streams/src/test/java/com/island/ohara/streams/ostream/TestPurchaseAnalysis.java [hunk 1]
```diff
@@ -137,10 +134,21 @@ public void testStreamApp() {
...
+                                Math.abs(
+                                        d
+                                            - record
+                                                .key()
+                                                .map(
+                                                    row ->
+                                                        Double.valueOf(
+                                                            row.cell("amount").value().toString()))
+                                                .orElse(-999.0))
+                                    < THRESHOLD));
```
Comment (vitojeng) [inline ohara-streams/src/test/java/com/island/ohara/streams/ostream/TestPurchaseAnalysis.java]: 這裡 code format 會不會太奇怪?

File: ohara-streams/src/test/java/com/island/ohara/streams/ostream/TestPurchaseAnalysis.java [hunk 2]
```diff
@@ -85,42 +85,39 @@ public void setup() {
...
-          .numberOfReplications(replications)
-          .topicName(orderuser_repartition)
-          .create();
     } catch (Exception e) {
       LOG.error(e.getMessage());
     }
 
     // write users.csv to kafka broker
-    produceData("users.csv", userTopic, "name");
+    //    produceData("users.csv", userTopic, "name");
```
Comment (vitojeng) [inline ohara-streams/src/test/java/com/island/ohara/streams/ostream/TestPurchaseAnalysis.java]: common code 應該可以拿掉?

File: ohara-streams/src/test/java/com/island/ohara/streams/ostream/TestPurchaseAnalysis.java [hunk 3]
```diff
@@ -85,42 +85,39 @@ public void setup() {
...
+    produceData("users.csv", userTopic);
 
     // write items.csv to kafka broker
-    produceData("items.csv", itemTopic, "itemName");
+    //    produceData("items.csv", itemTopic, "itemName");
```
Comment (vitojeng) [inline ohara-streams/src/test/java/com/island/ohara/streams/ostream/TestPurchaseAnalysis.java]: ditto

File: ohara-streams/src/test/java/com/island/ohara/streams/ostream/TestPurchaseAnalysis.java [hunk 4]
```diff
@@ -85,42 +85,39 @@ public void setup() {
...
+    produceData("items.csv", itemTopic);
 
     // write orders.csv to kafka broker
-    produceData("orders.csv", orderTopic, "userName");
+    //    produceData("orders.csv", orderTopic, "userName");
```
Comment (vitojeng) [inline ohara-streams/src/test/java/com/island/ohara/streams/ostream/TestPurchaseAnalysis.java]: ditto

File: ohara-streams/src/test/java/com/island/ohara/streams/ostream/TestPurchaseAnalysis.java (hunk 1 above)
Comment (saivirtue) [reply ohara-streams/src/test/java/com/island/ohara/streams/ostream/TestPurchaseAnalysis.java]: spotlessJava自己改的QQ
我試著要弄的"好看"一點都會被改掉...

File: ohara-streams/src/test/java/com/island/ohara/streams/ostream/TestPurchaseAnalysis.java (hunk 2 above)
Comment (saivirtue) [reply ohara-streams/src/test/java/com/island/ohara/streams/ostream/TestPurchaseAnalysis.java]: thanks :)

File: ohara-streams/src/test/java/com/island/ohara/streams/ostream/TestPurchaseAnalysis.java (hunk 1 above)
Comment (saivirtue) [reply ohara-streams/src/test/java/com/island/ohara/streams/ostream/TestPurchaseAnalysis.java]: 我嘗試把變數拉出來，讓format變的好看一點

**Pull request merged**